core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/regionserver/transactional/KeyValueListScanner.java.tmpl

```diff
@@ -48,7 +50,7 @@ public class KeyValueListScanner implements KeyValueScanner {
         Collections.sort(data, this.comparator);
         this.iter = data.iterator();
         if (!data.isEmpty()) {
-            this.current = KeyValue.createFirstOnRow(data.get(0).getRow());
+            this.current = KeyValueUtil.createFirstOnRow(CellUtil.cloneRow(data.get(0)));
```

Comment:
Please check cloning of the row here is absolutely needed. This is going to get very expensive memory consumption.